conftest.py

```diff
@@ -88,10 +88,22 @@ def pytest_generate_tests(metafunc):
         metafunc.parametrize(
             "container",
             [
-                (ver, containers[container_type][ver])
-                for ver in containers[container_type]
+                (
+                    container.tag,
+                    "/".join(
```

Comment:
this should be container.url, and the url should be done automatically, only once.